dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,626 @@
+/*
+    DSSCollection contains collection group & completion functionality for DSS.
+    Author: Jeremy Ahrens jer.ahrens@dapperlabs.com
+*/
+
+import NonFungibleToken from "./NonFungibleToken.cdc"
+import MetadataViews from 0xMETADATAVIEWSADDRESS
+
+// The DSSCollection contract
+//
+pub contract DSSCollection: NonFungibleToken {
+
+    // Contract Events
+    //
+    pub event ContractInitialized()
+
+    // NFT Collection Events
+    //
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+
+    // Events
+    //
+    pub event CollectionGroupCreated(
+        id: UInt64,
+        name: String,
+        typeName: String,
+        startTime: UFix64?,
+        endTime: UFix64?,
+        timeBound: Bool
+    )
+    pub event CollectionGroupClosed(id: UInt64)
+    pub event SlotCreated(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        logicalOperator: String,
+        typeName: String,
+        slotType: String
+    )
+    pub event ItemAddedToSlot(
+        itemID: UInt64,
+        itemValue: UInt64,
+        collectionGroupID: UInt64
+    )
+    pub event DSSCollectionNFTMinted(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        serialNumber: UInt64,
+        completedBy: String,
+        completionDate: UFix64,
+        level: UInt64
+    )
+    pub event DSSCollectionNFTBurned(id: UInt64)
+
+
+    // Named Paths
+    //
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let AdminStoragePath:       StoragePath
+    pub let MinterPrivatePath:      PrivatePath
+
+    // Entity Counts
+    //
+    pub var totalSupply:                 UInt64
+    pub var nextCollectionGroupID:       UInt64
+    pub var nextSlotID:       UInt64
+
+    // Lists in contract
+    //
+    access(self) let collectionGroupByID: @{UInt64: CollectionGroup}
+    access(self) let slotByID: @{UInt64: Slot}
+
+    // A public struct to access Slot data
+    //
+    pub struct SlotData {
+        pub let id: UInt64
+        pub let collectionGroupID: UInt64
+        pub let logicalOperator: String // (AND / OR)
+        pub let typeName: String // (A.contractAddress.NFT...)
+        pub let slotType: String // (edition.id, edition.tier, play.id)
+        pub var items: {UInt64: AnyStruct}
```

Comment:
What exactly are the Items?

Reply:
Items can be anything from an edition.id, play.id, or nft.id. A rarity, which is an int, etc.

In product terms, could be a specific Lebron Moment, a rare moment, or a specific play.id.

Reply:
So the slots will be used to determine which NFTs form a certain CollectionGroup, but there will be more than one Slot per CollectionGroup? If I'm understanding correctly, a `Slot.slotType` could be == `play.id` and then it will have as many items as plays we want to be in the CollectionGroup?

Reply:
Yes, a collection-group is made up of slots & the slots have rules within them so that game designers can make flexible games within the collection group. The collection-group mints a token to the user when they have completed the collection-group requirements. 

Each slot has a LogicalOperator, AND/OR, and is made up of items, which can be edition.Ids, play.ids, rarity.ids, etc. Each item has points associated with it, so if you collect an edition.id that is more valuable in the game, you will get more points.

Then, the points you have are summed, and the BE application applies a level to the NFT minted to your wallet when you complete.

Hope that helps. Appreciate you reviewing the contract!